lib/rspec/abq/extensions.rb

```diff
@@ -125,11 +125,15 @@ def run_specs(example_groups)
             end
           end
 
-          if Abq.target_test_case == Abq::TestCase.end_marker
+          if Abq.target_test_case != Abq::TestCase.end_marker
+            warn "Hit end of test run without being on end marker. Target test case is #{Abq.target_test_case.inspect}"
+            example_passed = false
+          end
+
+          if Gem::Version.new(RSpec::Core::Version::STRING) >= Gem::Version.new("3.10.0")
```

Comment:
What happened here such that we didn't need this logic branch before?

Reply:
we started surfacing errors in `abq work` that were swallowing these errors for old versions